src/software/visualizer/widgets/ai_control.cpp

```diff
@@ -110,40 +110,39 @@ void setupGameStateOverrideComboBox(Ui::AutoGeneratedMainWidget *widget)
                      on_gamestate_changed);
 }
 
-void setupDefendingSideComboBox(Ui::AutoGeneratedMainWidget *widget)
+void setupDefendingSideComboBox(Ui::AutoGeneratedMainWidget *widget,
+                                std::shared_ptr<ThunderbotsConfig> config)
 {
     // See issue #811 for getting these value from an enum / factory
     widget->defending_side_combo_box->insertItem(0, "Use Refbox");
     widget->defending_side_combo_box->insertItem(1, "East");
     widget->defending_side_combo_box->insertItem(2, "West");
 
-    auto on_defending_side_changed = [](const QString &text) {
+    auto on_defending_side_changed = [config](const QString &text) {
         if (text == "Use Refbox")
         {
-            Util::MutableDynamicParameters->getMutableAIControlConfig()
+            config->getMutableAIControlConfig()
```

Comment:
Similar comment here, could add `auto refbox_config = ...` at the top of this lambda and clean it up a bit, but your call.

Reply:
done